Simplified/NYPLBarcode.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import ZXingObjC
+
+fileprivate let barcodeHeight: CGFloat = 100
+
+@objc enum NYPLBarcodeType: Int {
+  case codabar
+  case code39
+  case qrCode
+  case code128
+}
+
+fileprivate func ZXBarcodeFormatFor(_ NYPLBarcodeType:NYPLBarcodeType) -> ZXBarcodeFormat {
+  switch NYPLBarcodeType {
+  case .codabar:
+    return kBarcodeFormatCodabar
+  case .code39:
+    return kBarcodeFormatCode39
+  case .qrCode:
+    return kBarcodeFormatQRCode
+  case .code128:
+    return kBarcodeFormatCode128
+  }
+}
+
+/// Manage creation of barcode images that are scannable by physical scanners.
+/// Keep any third party dependency abstracted out of the main app.
+final class NYPLBarcode: NSObject {
+
+  class func image(string: String, size: CGSize, type: NYPLBarcodeType) -> UIImage?
+  {
+    let writer = ZXMultiFormatWriter.writer() as? ZXWriter
+    do {
+      let result = try writer?.encode(string, format: ZXBarcodeFormatFor(type), width: Int32(size.width), height: Int32(size.height))
+
+      if let cgImage = ZXImage.init(matrix: result).cgimage {
+        return UIImage.init(cgImage: cgImage)
+      } else {
+        Log.error(#file, "Error creating image from barcode string.")
+        return nil
+      }
+    } catch {
+      Log.error(#file, "Failed to create barcode image with error: \(error.localizedDescription)")
+      return nil
+    }
+  }
+
+  class func size(forSuperviewBounds bounds: CGRect) -> CGSize
```

Comment:
This seems like a strange method to expose, perhaps. Maybe the `image` method should take a `maxSize` argument instead of `size` and this logic should be kept internal to the class?

Reply:
Good call. I've internalized that method, and changed the signature to ask for the superview width so I can make a determination (internally) for a good width depending on the screen, and fixing the height as a constant also defined internally.